src/org/mozilla/javascript/ScriptRuntime.java

```diff
@@ -4293,22 +4338,30 @@ public static Scriptable newObjectLiteral(
             // -1 for property getter, 1 for property setter, 0 for a regular value property
             int getterSetter = getterSetters == null ? 0 : getterSetters[i];
             Object value = propertyValues[i];
+
             if (getterSetter == 0) {
-                if (id instanceof String) {
-                    if (isSpecialProperty((String) id)) {
-                        Ref ref = specialRef(object, (String) id, cx, scope);
+                if (id instanceof Symbol) {
+                    Symbol sym = (Symbol) id;
+                    SymbolScriptable so = (SymbolScriptable) object;
+                    so.put(sym, object, value);
+                } else if (id instanceof Integer) {
+                    int index = ((Integer) id).intValue();
+                    object.put(index, object, value);
+                } else {
+                    String stringId = ScriptRuntime.toString(id);
+                    if (isSpecialProperty(stringId)) {
+                        Ref ref = specialRef(object, stringId, cx, scope);
                         ref.set(cx, scope, value);
                     } else {
-                        object.put((String) id, object, value);
+                        object.put(stringId, object, value);
                     }
-                } else {
-                    int index = ((Integer) id).intValue();
-                    object.put(index, object, value);
                 }
             } else {
                 ScriptableObject so = (ScriptableObject) object;
                 Callable getterOrSetter = (Callable) value;
                 boolean isSetter = getterSetter == 1;
+                // XXX: Do we have to handle Symbol here.
+                // This will be required, when conputedprops are supported.
```

Comment:
Yes, we need to check this. See https://github.com/FOCONIS/rhino/commit/6d2a459e0d44f8efc59a0b1498da56beec1d2e9c
But I think this should be done in a separate PR